src/betterproto/__init__.py

```diff
@@ -169,7 +169,22 @@ class Casing(builtin_enum.Enum):
     SNAKE = snake_case  #: A snake_case sterilization function.
 
 
-PLACEHOLDER: Any = object()
+class Placeholder:
+    __slots__ = ()
+
+    def __repr__(self) -> str:
+        return "<PLACEHOLDER>"
+
+    def __copy__(self) -> Self:
+        return self
+
+    def __deepcopy__(self, _) -> Self:
+        return self
+
+
+# We can't simply use object() here because pydantic automatically performs deep-copy of mutable default values
+# See #606
+PLACEHOLDER: Placeholder = Placeholder()
```

Comment:
```suggestion
PLACEHOLDER = cast(Any, Placeholder())
```

Reply:
I'm not sure to understand, is there any reason for this?

Reply:
It's so it can be used anywhere and the type checker won't complain

Reply:
I think the type checker complains anyway: with mypy, your changes turns:
`src/betterproto/__init__.py:224: error: Argument "default" to "field" has incompatible type "Placeholder | None"; expected "Field[Any]"  [arg-type]`
into
`src/betterproto/__init__.py:224: error: Argument "default" to "field" has incompatible type "Any | None"; expected "Field[Any]"  [arg-type]`

It is coherent since if x has type `T` and y has type `Any`, then `x = y` should not be a valid operation. Wouldn't it be better to add `# type: ignore` on the lines where it is needed?

Reply:
It should still be an Any

Reply:
I updated the PR